src/main/java/teammates/common/util/ActivityLogEntry.java

```diff
@@ -1,365 +1,310 @@
-        return role.replace(Const.ActivityLog.ROLE_MASQUERADE_POSTFIX, "").trim();
+    public long getLogTime() {
+        return logTime;
     }
 
-    public boolean isMasqueradeUserRole() {
-        return role.contains(Const.ActivityLog.ROLE_MASQUERADE_POSTFIX);
+    public long getActionTimeTaken() {
+        return actionTimeTaken;
     }
 
-    public String getRole() {
-        return role;
+    public String getActionName() {
+        return actionName;
     }
 
-    public String getName() {
-        return name;
+    public String getActionResponse() {
+        return actionResponse;
     }
 
-    public String getGoogleId() {
-        return googleId;
+    public String getUserRole() {
+        return userRole;
     }
 
-    public String getEmail() {
-        return email;
+    public boolean isMasqueradeUserRole() {
+        return isMasqueradeUserRole;
     }
 
-    public long getTimeTaken() {
-        return timeTaken;
+    public String getUserName() {
+        return userName;
     }
 
-    public String getUrl() {
-        return url;
+    public String getUserGoogleId() {
+        return userGoogleId;
     }
 
-    public String getMessage() {
-        return message;
+    public String getUserEmail() {
+        return userEmail;
     }
 
-    /**
-     * Generates the ID for the log. If the googleId is unknown or unregistered,
-     * the email and course of the {@code student} will be used to construct the id.
-     * @param googleId the google ID
-     * @param student StudentAttributes object
-     * @return log ID
-     */
-    public String generateLogId(String googleId, StudentAttributes student, long time) {
-        return student == null ? generateLogId(googleId, null, null, time)
-                               : generateLogId(googleId, student.email, student.course, time);
+    public boolean isTestingData() {
+        return userEmail.endsWith(Const.ActivityLog.TESTING_DATA_EMAIL_POSTFIX);
     }
 
     /**
-     * Generates the ID for the log. If the googleId is unknown or unregistered,
-     * the {@code email} and {@code course} will be used to construct the id.
-     * @param googleId the google ID
-     * @param email the email
-     * @param course the course
-     * @return log ID
+     * A builder class for {@link ActivityLogEntry}.
+     *
+     * <p>All optional fields are initialized to {@link Const.ActivityLog.UNKNOWN}.
+     * Field actionResponse will be initialized to have the same value as actionName.
+     * All null values (if possible) that are passed into the builder will be ignored.
+     *
+     * @see ActivityLogEntry
      */
-    public String generateLogId(String googleId, String email, String course, long time) {
-        boolean isUnregisteredStudent = (googleId.contentEquals("Unknown") || googleId.contentEquals("Unregistered"))
-                                        && email != null && course != null;
+    public static class Builder {
```

Comment:
If we separate the builder to `ActivityLogEntryBuilder`, it means that we either need to make a bunch of setters or make the fields public in `ActivityLogEntry`. In either way, the `ActivityLogEntryBuilder` seems to be redundant. 

The reason to use builder is to make ActivityLogEntry immutable. Therefore, wee need to make it as an inner class to access the private fields.